src/Installer/finalizer/Program.cs

```diff
@@ -0,0 +1,251 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32;
+using Microsoft.Win32.Msi;
+
+if (args.Length < 3)
```

Comment:
We'll need is to replace the console logging and make sure we log this to a file. Probably can wrap everything in a try/catch to make sure the log file can be closed/flushed in a finally block. The log file is critical so that we can capture enough information when users collect setup logs for us to analyze.

It's important to make sure to include the pre/postamble as that tells us the log is complete.

It's line should include the matching prefix we currently get from DUTIL libraries to ensure all the setup logs are consistent so that they can be processed through any automation using analysis tools

Here's a sample of the finalizer log:

```
[25AC:28FC][2024-11-26T08:12:54]i000: === Logging started: 2024-11-26T08:12:54-08:00 ===
[25AC:28FC][2024-11-26T08:12:54]i000: Executable: C:\ProgramData\Package Cache\D1C8582115FC0B79532C8CB398BB99E0621C796A\finalizer.exe v6.4.2824.52403
[25AC:28FC][2024-11-26T08:12:54]i000: Setting target dependent to Microsoft.NET.Sdk,6.0.400,x64.
[25AC:28FC][2024-11-26T08:12:54]i000: Detecting installed SDK versions for 6.0.400
[25AC:28FC][2024-11-26T08:12:54]i000: Scanning SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\sdk
[25AC:28FC][2024-11-26T08:12:54]i000: === Logging stopped: 2024-11-26T08:12:54-08:00 ===
```
Each line is prefixed with `[Process ID:Thread ID][DateTime]`, and you can probably just do an extension method for StreamWriter and use `File.CreateText` to create the log file based on the command line argument that's passed, for example (you'll probably just need to capture the proc/thread ID in a static variable as it won't change.

```csharp
public static void Log(this StreamWriter sw, string message)
{
    var pid = Process.GetCurrentProcess().Id;
    var tid = Environment.CurrentManagedThreadId;
    sw.WriteLine($"[{pid:X4}:{tid:X4}][{DateTime.Now:yyyy-MM-ddTHH:mm:ss}] {message}");
}
 
 
 ```